R/assertions.R

```diff
@@ -1448,3 +1448,51 @@ assert_date_var <- function(dataset, var, dataset_name = NULL, var_name = NULL)
     ))
   }
 }
+
+#' Is a Variable is a Date or Datetime Variable?
+#'
+#' Checks if a variable vector is a date or datetime variable
+#'
+#' @param arg The function argument to be checked
+#'
+#' @param var_name a character vector name of the `arg`.
+#' `var_name` argument is optional.
+#'
+#' @param optional Is the checked parameter optional? If set to `FALSE`
+#' and `arg` is `NULL` then an error is thrown.
+#'
+#' @return
+#' The function returns an error if `arg` is not a date or datetime variable
+#' but otherwise returns an invisible output.
+#'
+#' @export
+#'
+#' @author Sadchla Mascary
+#'
+#' @keywords assertion
+#'
+#' @examples
+#' example_fun <- function(arg) {
+#'   assert_date_vector(arg)
+#' }
+#'
+#' example_fun(
+#'   as.Date("2022-01-30", tz = "UTC")
+#' )
+#' try(example_fun(c("2018-08-23", "2022-01-30", "1993-07-14")))
+assert_date_vector <- function(arg, var_name = NULL, optional = FALSE) {
+  assert_character_vector(var_name, optional = TRUE)
+  assert_logical_scalar(optional)
```

Comment:
The logic for `optional` is missing, i.e. if `arg` is `NULL` and `optional = TRUE` then the function should exit early without even checking the input.

Reply:
got it, the logic has now being implemented